Assets/Scripts/Algorithms/Patrolling/HeuristicConscientiousReactive.cs

```diff
@@ -48,6 +48,8 @@ public sealed class HeuristicConscientiousReactiveAlgorithm : PatrollingAlgorith
         private GoToNextVertexComponent _goToNextVertexComponent = null!;
         private CollisionRecoveryComponent _collisionRecoveryComponent = null!;
 
+        private readonly System.Random _random = new(0);
```

Comment:
Should we give the Random a seed instead?

Reply:
Yep